pycloudlib/ec2/instance.py

```diff
@@ -129,7 +211,7 @@ def delete(self, wait=True) -> List[Exception]:
         """Delete instance."""
         exceptions = []
         # Even with DeleteOnTermination set True, nics can outlive instances
-        for ip in self.created_interfaces:
+        for ip in self.created_interfaces[:]:
```

Comment:
Why have you opted for this syntax?

Reply:
I am creating a copy of that list to iterate over because I want to remove the created interfaces from the list while being removed from this loop or manually.

See:

https://github.com/canonical/pycloudlib/pull/355/files#diff-0c9a2ff75eae3b564b5813cf4f5d5ea8cb15fbbf3b680b780fcb102503dc7e5cR550

Reply:
Okay I had not seen that.